### ISSUE TITLE
kakoune: Unicode characters are broken on recent version

### ISSUE THREAD

**Issue by hansjoergschurr** [NixOS/nixpkgs#52725]
## Issue description
When using a kakoune newer than `kakoune-unstable-2018-02-15` such as the current `kakoune-unstable-2018.10.2` the printing of Unicode characters is broken. This includes the border decoration of the help windows.

For example when opening this file in Kakoune: `https://raw.githubusercontent.com/sharpsaw/unicode-dots/master/eg/UTF-8-demo.txt` instead of the Unicode characters an invisible two space wide character is printed. When opened in Vim it works just fine. This behavior is consistent in multiple terminal emulators. 

`i8n.defaultLocale` is set to `en_CA.UTF-8` and running `locale` produces:
```
LANG=en_CA.UTF-8
LC_CTYPE="en_CA.UTF-8"
LC_NUMERIC="en_CA.UTF-8"
LC_TIME="en_CA.UTF-8"
LC_COLLATE="en_CA.UTF-8"
LC_MONETARY="en_CA.UTF-8"
LC_MESSAGES="en_CA.UTF-8"
LC_PAPER="en_CA.UTF-8"
LC_NAME="en_CA.UTF-8"
LC_ADDRESS="en_CA.UTF-8"
LC_TELEPHONE="en_CA.UTF-8"
LC_MEASUREMENT="en_CA.UTF-8"
LC_IDENTIFICATION="en_CA.UTF-8"
LC_ALL=
```

### Steps to reproduce
Install and run kakoune

## Technical details

 - system: `"x86_64-linux"`
 - host os: `Linux 4.14.76, NixOS, 18.03.133389.b551f89e256 (Impala)`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.0.4`
 - channels(root): `"nixos-18.03.133389.b551f89e256"`
 - channels(hans): `"stable-18.03.133389.b551f89e256, unstable-19.03pre162671.44b02b52ea6"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos/nixpkgs`

**Comment by hansjoergschurr** [comment]
This seems to be the same problem as reported in the kakoune repository here: https://github.com/mawww/kakoune/issues/2471

**Comment by hansjoergschurr** [comment]
After upgrading everything to 18.09 the problem disappeared. As discussed in the referenced issue (and issues referenced there) the problem might have been related to the glibc version. 
`strace kak` printed:
````
open("/nix/store/hwwqshlmazzjzj7yhrkyjydxamvvkfd3-glibc-2.26-131/lib/locale/en_CA.UTF-8/LC_TIME", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
open("/nix/store/hwwqshlmazzjzj7yhrkyjydxamvvkfd3-glibc-2.26-131/lib/locale/en_CA.utf8/LC_TIME", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
open("/nix/store/hwwqshlmazzjzj7yhrkyjydxamvvkfd3-glibc-2.26-131/lib/locale/en_CA/LC_TIME", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
````
while 
````
$ echo $LOCALE_ARCHIVE 
/run/current-system/sw/lib/locale/locale-archive
$ file -d /run/current-system/sw/lib/locale
/run/current-system/sw/lib/locale: symbolic link to /nix/store/cah17w88csqwgaicdkjs65w8g0ss3yq1-glibc-locales-2.27/lib/locale
````

While this solves the problem for me, I'm not sure if this is enough to close the issue, since the problem persists for users wanting to use the 18.03 and 18.09 channel at the same time.

**Comment by eraserhd** [comment]
@hansjoergschurr I think this can be closed, since we are on 19.09.

**Comment by hansjoergschurr** [comment]
Sure!
